test/e2e/framework/util.go

```diff
@@ -2757,26 +2761,87 @@ func WaitForPodsWithLabel(c clientset.Interface, ns string, label labels.Selecto
 	return
 }
 
-// DeleteRCAndPods a Replication Controller and all pods it spawned
-func DeleteRCAndPods(clientset clientset.Interface, internalClientset internalclientset.Interface, ns, name string) error {
-	By(fmt.Sprintf("deleting replication controller %s in namespace %s", name, ns))
-	rc, err := clientset.Core().ReplicationControllers(ns).Get(name)
+func getSelectorAndReplicasFromResource(c clientset.Interface, resource, ns, name string) (*unversioned.LabelSelector, int32, error) {
```

Comment:
Instead of passing "string" resource, we should use "kind schema.GroupKind" - see e.g. for example usage
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/history.go#L46